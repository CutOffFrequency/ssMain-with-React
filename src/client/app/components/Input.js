import React from "react";
import { passSelector} from "../services/";

const Input = props => {
  let {
    selector,
    prompt,
    value,
    change,
    submit,
    messageSelector,
    message
  } = props;

  return (
    <div className={passSelector"div")(selector)}>
      <p className={passSelector"p")(selector)}>{prompt}</p>
      <div className={passSelector"innerDiv")(selector)}>
        <input
          className={passSelector"input")(selector)}
          type="text"
          value={value}
          onChange={change}
        />
        <button className={passSelector"button")(selector)} onClick={submit}>
          Submit
        </button>
        <p className={messageSelector}>{message}</p>
      </div>
    </div>
  );
};

export default Input;
