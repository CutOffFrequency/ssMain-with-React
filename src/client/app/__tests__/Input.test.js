import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Input from '../Input.react';

it('should render', () => {
    const component = renderer.create(
        <Input selector="accountInput" text="submit account"
        />
    );
    // initial state
    let tree = component.toJSON();
    // compare to snapshot
    expect(tree).toMatchSnapshot();
})

it('className should match prop.className', () => {
    const component = renderer.create(
        <Input selector="accountInput" text="submit account"
        />
    );
    // initial state
    let tree = component.toJSON();
    // compare to snapshot
    expect(tree).toMatchSnapshot();
    // find input
    const inp = tree.children.filter(child => child.type === 'input')[0];
    expect(inp.props.className).toBe("accountInput input");
})