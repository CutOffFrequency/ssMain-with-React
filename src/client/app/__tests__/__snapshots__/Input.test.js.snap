// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`className should match prop.className 1`] = `
<div
  className="accountInput-parentDiv"
>
  <p
    className="accountInput p"
  />
  <input
    className="accountInput input"
    onChange={[Function]}
    value={undefined}
  />
  <button
    className="accountInput button"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;

exports[`should render 1`] = `
<div
  className="accountInput-parentDiv"
>
  <p
    className="accountInput p"
  />
  <input
    className="accountInput input"
    onChange={[Function]}
    value={undefined}
  />
  <button
    className="accountInput button"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
